src/pages/index.js

```diff
@@ -21,80 +21,271 @@ import {
-  return card;
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/cohort-3-en",
+  headers: {
+    authorization: "6960473d-b35e-4d3c-8ef0-2e694acf306b",
+    "Content-Type": "application/json",
+  },
+});
+
+let userId;
+let cardListSection;
+
+Promise.all([api.getInitialCards(), api.getUserInfo()])
+  .then(([initialCards, userData]) => {
+    userId = userData._id;
+    console.log(userId);
+    userInfo.setUserInfo(userData.name, userData.about);
+    userInfo.setProfileAvatar(userData.avatar);
+    cardListSection = new Section(
+      {
+        items: initialCards,
+
+        renderer: ({ name, link, likes, _id, owner }) => {
+          const newCard = createCard({
+            name,
+            link,
+            likes,
+            _id,
+            userId,
+            ownerId: owner._id,
+          });
+          cardListSection.addItem(newCard);
+        },
+      },
+      cardList
+    );
+    cardListSection.renderItems();
+  })
+  .catch(() => (err) => console.log(err));
+//--------------------------------------------------------------AVATAR-------------------------------------------------------------------------------------
+//new avatar constants
+
+const profileAvatarModalImageSelector = document.querySelector(
+  // "#update-avatar-modal"
+  ".profile__image"
+);
+const profileAvatarModalSelector = document.querySelector(
+  "#update-avatar-modal"
+);
+const profileAvatarPeniclIcon = document.querySelector(
+  ".profile__icon-edit-button"
+);
+
+//avatar form validator instance
+
+const avatarFormValidator = new FormValidator(
+  settings,
+  profileAvatarModalSelector
+);
+avatarFormValidator.enableValidation();
+
+//edit avatar popup instance
+
+const editAvatarPopup = new PopupWithForm(
+  profileAvatarModalSelector,
+  handleAvatarFormSubmit
+);
+editAvatarPopup.setEventListeners();
+
+// avatar form submit function
+
+function handleAvatarFormSubmit({ url }) {
+  editAvatarPopup.setLoading(true);
+  api
+    .setUserAvatar(url)
+    .then((userData) => {
+      userInfo.setProfileAvatar(userData.avatar);
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editAvatarPopup.setLoading(false, "Save");
+    });
 }
 
-//FORM VALIDATOR
+//calling avatar popup
 
-const addCardFormValidator = new FormValidator(settings, profileAddModal);
-addCardFormValidator.enableValidation();
+profileAvatarPeniclIcon.addEventListener("click", () => {
+  editAvatarPopup.open();
+});
 
-const editProfileFormValidator = new FormValidator(settings, profileEditModal);
-editProfileFormValidator.enableValidation();
+//--------------------------------------------------DELETE-----------------------------------------------------------------------------------
+// new delete card const
 
-//SECTION TO RENDER CARDS
+const deleteCardModalSelector = document.querySelector("#delete-confirm-modal");
```

Comment:
* Word `selector` means a **string** that is used for searching elements in the DOM